File: array.reverse/reverse.cpp

#include <iostream>


template<typename T_type, int T_length>
static void print_array(const T_type (&array)[T_length])
{
	for(int i = 0; i < T_length; i++)
		std::cout << array[i] << " ";
	std::cout << std::endl;
}

template<typename T_type>
static void swap(T_type& v1, T_type& v2)
{
	T_type temp = v1;
	v1 = v2;
	v2 = temp;
}

template<typename T_type, int T_length>
static void reverse(T_type (&array)[T_length])
{
	int half_length = T_length >> 1;
	for(int i = 0; i < half_length; i++)
		swap(array[i], array[T_length - i - 1]);
}

int main()
{
	int array[] = { 4, 5, 6, -2, 3, -5, 56, 7 };
	std::cout << "Original Array: \n";
	print_array(array);
	reverse(array);
	std::cout << "Reversed Array: \n";
	print_array(array);

	return 0;
}

File: array.reverse/reverse.py

def printArray(array):
	for val in array:
		print('{0} '.format(val), end ="")
	print('')
	return

def swap(v1, v2):
	return v2, v1

def reverse(array):
	length = len(array)
	halfLength = length >> 1
	for i in range(halfLength):
		array[i], array[length - i - 1] = swap(array[i], array[length - i - 1])
	return

def main():
	array = [ 5, 6, 7, 100, -10, 223, -34 ]
	print('Original Array')
	printArray(array)
	reverse(array)
	print('Reversed Array')
	printArray(array)

main()
